lib/AST/ASTContext.cpp

```diff
@@ -3266,18 +3266,18 @@ BoundGenericType *BoundGenericType::get(NominalTypeDecl *TheDecl,
 
   BoundGenericType *newType;
   if (auto theClass = dyn_cast<ClassDecl>(TheDecl)) {
-    auto mem = C.Allocate(sizeof(BoundGenericClassType) + sizeof(Type) *
-                          GenericArgs.size(), alignof(Type), arena);
+    auto sz = BoundGenericClassType::totalSizeToAlloc<Type>(GenericArgs.size());
+    auto mem = C.Allocate(sz, alignof(BoundGenericClassType), arena);
```

Comment:
This is an incorrect change; Type is analogous to `BoundGenericClassType *` not to `BoundGenericClassType` itself.

Reply:
The change is correct. Type is analogous to TypeBase*, and therefore alignof(Type) == alignof(void*), which has nothing to do with the alignment of what is pointed / being allocated.

Reply:
In short, you're confusing alignof(pointer) with alignof(pointee).

Reply:
Whoops, I thought *you* had confused alignof(pointer) with alignof(pointee) because I thought this was about the alignment of trailing objects. Sorry for the noise!

Reply:
No problem-o. I tripped over this bug while experimenting with a "compressed" `Type` where `sizeof(Type)` == `alignof(Type)` == `4`. I haven't finished that experiment yet due to other assumptions in the system (see my other commit today), but I thought that I should at least clean up the problems I tripped over along the way. :-)